Add "-c" arg when expecting llvm bitcode output

FantasqueX opened the pull request:
If there isn't "-c" argument, clang will generate an error "clang-12: error: -emit-llvm cannot be used when linking".

banach-space commented:
Great catch, thank you! Would you mind updating https://github.com/banach-space/llvm-tutor/blob/ad7dddf9affd4d5f218c934db7e698bb747ee2ba/tools/StaticMain.cpp#L12 as well?

-Andrzej

FantasqueX commented:
OK, I have updated StaticMain.cpp. And I think the text below the code I changed also needs to be changed, "Similar logic applies to opt (by default it generates *.bc files, use -S to generate an *.ll file instead).". "by default" is not accurate, but because I'm not a native English speaker, I don't know how to modify it appropriately.

banach-space commented:
Thank you for updating this!

> "by default" is not accurate

Good point, I should refine that. I'm not a native speaker myself and this is a bit tricky to explain. Let me take care of that!

banach-space commented:
Please, could you squash?

FantasqueX commented:
I have a question about why not to use "Squash and merge". I think it's a method more convenient than using "rebase". I haven't sent many pull requests before, so I'm curious to know why.

banach-space commented:
> why not to use "Squash and merge". 

When I "squash and merge" then it's up to me to either:

1. Leave your commit messages as is and simply combine them. But that would mean the following:
```
Add "-c" arg when expecting llvm bitcode output
Add '-c' arg in StaticMain.cpp
```
This is not a particularly useful commit message though. In fact, it repeats the key information twice (i.e. `Add "-c"`). So I'd rather have it refined. This brings me to the next point.

2. Refine it on your behalf. But then I'm editing your work and this means that you are not 100% in charge of your own changes. 

Btw, I'm happy to "squash and merge" if that's easier for you! :) And I might be missing some GitHub feature here as well.

-Andrzej

banach-space commented:
Congrats on the 200th commit to **llvm-tutor** !

FantasqueX commented:
How to combine two commit messages is quite a problem. I think an elegant solution would be for me to send the issue first, and then send the pull request after discussion. I'll try this approach next time.

banach-space commented:
How do you rebase? I normally just run `git rebase -i main` and that usually works just fine. I imagine that that's how "Squash and merge" actually works on GitHub, but I'm not 100% sure.
